core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -186,8 +186,21 @@ private static void throwFailure(
     throw new RESTException("Unhandled error: %s", errorResponse);
   }
 
-  private URI buildUri(String path, Map<String, String> params) {
+  private String buildBaseUri(String path) {
```

Comment:
Nit: the method name here feels wrong, we're building the path, not the base URI